modules/message-lane/src/inbound_lane.rs

```diff
@@ -93,7 +108,21 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 		}
 
 		data.latest_received_nonce = nonce;
-		data.relayers.push_back((nonce, relayer));
+
+		if data
+			.relayers
+			.back()
+			.map(|(_, _, last_relayer)| *last_relayer == relayer)
+			.unwrap_or(false)
+		{
+			data.relayers
+				.back_mut()
+				.expect("already ensured back relayer exists in preceding check")
+				.1 = nonce;
+		} else {
+			data.relayers.push_back((nonce, nonce, relayer));
+		}
```

Comment:
Here I'd prefer a match as well, but I think you'll run into lifetime problems, so only something like this might work:
```rust
let push_new = match data.relayers.back_mut() {
  Some(ref mut (first_nonce, last_nonce, last_relayer)) if last_relayer == &relayer => {
    *last_nonce = nonce;
    false
  },
  _ => true,
};
if push_new {
  data.relayers.push_back((nonce, nonce, relayer));
}

Reply:
Addressed.